assembl/models/idea.py

```diff
@@ -999,11 +999,30 @@ class RootIdea(Idea):
 
     @property
     def num_posts(self):
-        """ In the root idea, this is the count of all mesages in the system """
+        """ In the root idea, num_posts is the count of all non-deleted mesages in the discussion """
+        from .post import Post
+        result = self.db.query(Post).filter(
+            Post.discussion_id == self.discussion_id,
+            Post.hidden==False,
+            Post.tombstone_condition()
+        ).count()
+        return int(result)
+
+    @property
+    def num_read_posts(self):
+        """ In the root idea, num_posts is the count of all non-deleted read mesages in the discussion """
         from .post import Post
```

Comment:
Thanks for this explanation. It makes things clearer. I do agree with you that in order to prevent circular import, it is better to avoid declaring imports at the beginning of the file. So the two options are either in the function bodies or at the end of the file. I do believe that you guys had good reasons to do it this way. So no problem for me to leave it like as is. Thanks guys.